src/cts/src/TechChar.cpp

```diff
@@ -1332,18 +1327,20 @@ void TechChar::updateBufferTopologies(TechChar::SolutionData& solution)
 
 std::vector<size_t> TechChar::getCurrConfig(const SolutionData& solution)
 {
+  std::stringstream tmp;
   std::vector<size_t> config;
   for (auto inst : solution.instVector) {
     size_t masterID = cellNameToID(inst->getMaster()->getName());
     config.emplace_back(masterID);
   }
 
   if (logger_->debugCheck(CTS, "tech char", 1)) {
-    std::cout << "currConfig: ";
+    tmp << "currConfig: ";
     for (unsigned i : config) {
-      std::cout << i << " ";
+      tmp << i << " ";
     }
-    std::cout << std::endl;
+    tmp << std::endl;
```

Comment:
Unnecessary as report() will add a new line.   Similarly in functions below.

Reply:
Fixed